File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] arr) {
        int n=arr.length;
        if (n==0) return 0;
        int longest=1;
        HashSet<Integer> st=new HashSet<>();
        for(int i=0;i<n;i++){
            st.add(arr[i]);
        }
        for (int it:st){
            if(!st.contains(it-1)){
                int cnt=1;
                int x=it;
                while(st.contains(x+1)){
                    x++;
                    cnt++;
                }
                longest=Math.max(longest,cnt);
            }
        }
        return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, arr: List[int]) -> int:
        n=len(arr)
        if n==0:
            return 0
        longest=1
        st=set(arr)
        for it in st:
            if(it-1) not in st:
                cnt=1
                x=it
                while(x+1) in st:
                    x+=1
                    cnt+=1
                longest=max(longest,cnt)
        return longest